File: api/Models/Team.cs
using System.Text.Json.Serialization;

namespace api.Models
{
    public class Team
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = default!;
        public string Color { get; set; } = default!;
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public virtual Season Season { get; set; } = default!;
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public Guid SeasonId { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<Driver>? Drivers { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
        public List<Result>? Results { get; set; }
    }
}

File: web/src/app/models/team.ts
import { Driver } from "./driver";
import { Result } from "./result";
import { Season } from "./season";

export class Team {
  constructor(
    public id: string,
    public name: string,
    public color: string,
    public season: Season,
    public drivers: Driver[],
    public results: Result[]
  ) {}
}
